marin/export/hf_upload.py

```diff
@@ -0,0 +1,259 @@
+import dataclasses
+import functools
+import logging
+import os
+import tempfile
+import time
+from dataclasses import dataclass
+from functools import wraps
+
+import fsspec
+import humanfriendly
+from fsspec.implementations.local import LocalFileSystem
+from huggingface_hub import create_commit, upload_folder
+from tqdm_loggable.auto import tqdm
+
+from marin.execution import ExecutorStep, InputName
+from marin.utils import fsspec_glob
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class UploadToHfConfig:
+    input_path: str | InputName
+    repo_id: str
+    repo_type: str = "dataset"
+    token: str | None = None
+    revision: str | None = None
+
+    upload_kwargs: dict[str, str] = dataclasses.field(default_factory=dict)
+    """Will be passed to huggingface_hub.upload_folder"""
+    private: bool = False
+    commit_batch_size: str = "1GiB"
+    small_file_limit: str = "5 MiB"
+    """
+    The size limit for small files. Files larger than this will be uploaded using lfs.
+    """
+
+
+def upload_dir_to_hf(
+    input_path: str | InputName,
+    repo_id: str,
+    repo_type: str = "dataset",
+    token: str | None = None,
+    certificate_path: str | None = None,
+    private: bool = False,
+    revision: str | None = None,
+    commit_batch_size: str = "1GiB",
+    **upload_kwargs: str,
+) -> ExecutorStep:
+    """
+    Uploads a path (possibly a GCS path) to a Hugging Face repo.
+    For local paths, it will use the huggingface_hub.upload_folder function. For GCS (or other fsspec paths),
+    it will stream the files using preupload_lfs_files and/or upload_folder
+
+    Args:
+        input_path: path to upload (can be a GCS path)
+        repo_id: the repo id to upload to (e.g. "username/repo_name")
+        repo_type: the type of repo to upload to (e.g. "dataset", "model", etc.)
+        token: the token to use for authentication (if not provided, it will use the default token)
+        revision: the branch to upload to (if not provided, it will use the default branch)
+        certificate_path: where to store the certificate that we uploaded to HF (needed for executor idempotency).
+           If not provided, a reasonable default will be used.
+    Returns:
+        ExecutorStep
+    """
+    if not certificate_path:
+        certificate_path = f"metadata/hf_uploads/{input_path.name}"
+
+    return ExecutorStep(
+        name=certificate_path,
```

Comment:
if certificate_path is a full path, this might not make sense

Reply:
clarified it should be a relative path